sklearn/callback/tests/_utils.py

```diff
@@ -37,7 +37,37 @@ def _on_fit_task_end(self, estimator, context, **kwargs):
         pass  # pragma: no cover
 
 
-class Estimator(CallbackSupportMixin, BaseEstimator):
+class BaseEstimatorPrivateFit(BaseEstimator):
+    """A class that adds the implementation of a public and private fit method to the
+    BaseEstimator class.
+    """
+
+    @_fit_context(prefer_skip_nested_validation=False)
+    def fit(self, X_train=None, y_train=None, X_val=None, y_val=None):
```

Comment:
That's not going to change :)
```suggestion
    def fit(self, X=None, y=None, X_val=None, y_val=None):
```